File: client/web/src/features/audit/api/audit-api.ts
import { apiClient } from "@/lib/api-client";
import type { AuditEvent } from "@/types/api-types";

export interface AuditEventsResponse {
  events: AuditEvent[];
}

export function fetchAuditEvents(limit: number = 200): Promise<AuditEventsResponse> {
  return apiClient.get<AuditEventsResponse>(`/security/audit-events?limit=${limit}`);
}

File: client/android/core/src/main/java/io/github/pstanar/pstotp/core/api/AuditApi.kt
package io.github.pstanar.pstotp.core.api

import io.github.pstanar.pstotp.core.model.api.AuditEventListResponse

/** Audit event API endpoints (authenticated). */
class AuditApi(private val client: ApiClient) {

    suspend fun fetchEvents(limit: Int = 50): AuditEventListResponse =
        client.get("/security/audit-events?limit=$limit") { AuditEventListResponse.fromJson(it) }
}
